src/plugins/dashboard/public/locator.ts

```diff
@@ -95,6 +103,10 @@ export interface DashboardAppLocatorDependencies {
   getDashboardFilterFields: (dashboardId: string) => Promise<Filter[]>;
 }
 
+export type ForwardedDashboardState = Omit<DashboardAppLocatorParams, 'panels'> & {
```

Comment:
why do we need this interface ? seems to be same as DashboardAppLocatorParams except for panels which are defined a bit differently but very similarly ?

Reply:
We could keep it them same, but I think it communicates, semantically, what value the app expects to receive internally on `history.location.state`.

An alternative is to pass `panels` to the app as is and the "deserialize" them. I'd like to get @ThomThomson 's thoughts too!